backends/c-sycl_intel/include/backend.hpp

```diff
@@ -2,12 +2,15 @@
 // Licensed under the MIT License.
 
 //; eval_flags(c-sycl_intel): [dpcpp] -ldl -lpthread
-//; eval_flags(c-sycl_cuda): [/usr/local/dpcpp-cuda/bin/clang++] -ldl -I/usr/local/dpcpp-cuda/include/sycl -L/usr/local/dpcpp-cuda/lib -lsycl -fsycl -fsycl-targets=nvptx64-nvidia-cuda-sycldevice -fsycl-unnamed-lambda -lpthread
+//; eval_flags(c-sycl_cuda): [/usr/local/dpcpp-cuda/bin/clang++] -ldl -I/usr/local/dpcpp-cuda/include/sycl -L/usr/local/dpcpp-cuda/lib -lsycl -fsycl -fsycl-targets=nvptx64-nvidia-cuda-sycldevice -fsycl-unnamed-lambda -lpthread -iquote/usr/local/cuda/include -L/usr/local/cuda/lib64 -lcuda -DSYCL_CUDA
 
 #include <CL/sycl.hpp>
 #include <dlfcn.h>
 #include <pthread.h>
 #include <malloc.h>
+#ifdef SYCL_CUDA
+#include "cuda.h"
+#endif
```

Comment:
Besides, can you make it from `"cuda.h"` to `<cuda.h>`?

Reply:
That is the issue I talked about before, if I move to `<cuda.h>`, I have to change the include from `-iquote` to `-I`, but then dpc++ picks up on the CUDA headers and I end up with:
```
In file included from /home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/exception.hpp:15:
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:251:7: error: use of undeclared identifier 'CL_DEVICE_QUEUE_ON_DEVICE_PROPERTIES'
      CL_DEVICE_QUEUE_ON_DEVICE_PROPERTIES,
      ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:252:45: error: use of undeclared identifier 'CL_DEVICE_QUEUE_ON_HOST_PROPERTIES'
  PI_DEVICE_INFO_QUEUE_ON_HOST_PROPERTIES = CL_DEVICE_QUEUE_ON_HOST_PROPERTIES,
                                            ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:256:31: error: use of undeclared identifier 'CL_DEVICE_IL_VERSION_KHR'
  PI_DEVICE_INFO_IL_VERSION = CL_DEVICE_IL_VERSION_KHR,
                              ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:274:39: error: use of undeclared identifier 'CL_DEVICE_MAX_NUM_SUB_GROUPS'
  PI_DEVICE_INFO_MAX_NUM_SUB_GROUPS = CL_DEVICE_MAX_NUM_SUB_GROUPS,
                                      ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:276:7: error: use of undeclared identifier 'CL_DEVICE_SUB_GROUP_INDEPENDENT_FORWARD_PROGRESS'
      CL_DEVICE_SUB_GROUP_INDEPENDENT_FORWARD_PROGRESS,
      ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:277:42: error: use of undeclared identifier 'CL_DEVICE_SUB_GROUP_SIZES_INTEL'
  PI_DEVICE_INFO_SUB_GROUP_SIZES_INTEL = CL_DEVICE_SUB_GROUP_SIZES_INTEL,
                                         ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:278:37: error: use of undeclared identifier 'CL_DEVICE_HOST_MEM_CAPABILITIES_INTEL'
  PI_DEVICE_INFO_USM_HOST_SUPPORT = CL_DEVICE_HOST_MEM_CAPABILITIES_INTEL,
                                    ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:279:39: error: use of undeclared identifier 'CL_DEVICE_DEVICE_MEM_CAPABILITIES_INTEL'
  PI_DEVICE_INFO_USM_DEVICE_SUPPORT = CL_DEVICE_DEVICE_MEM_CAPABILITIES_INTEL,
                                      ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:281:7: error: use of undeclared identifier 'CL_DEVICE_SINGLE_DEVICE_SHARED_MEM_CAPABILITIES_INTEL'
      CL_DEVICE_SINGLE_DEVICE_SHARED_MEM_CAPABILITIES_INTEL,
      ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:283:7: error: use of undeclared identifier 'CL_DEVICE_CROSS_DEVICE_SHARED_MEM_CAPABILITIES_INTEL'
      CL_DEVICE_CROSS_DEVICE_SHARED_MEM_CAPABILITIES_INTEL,
      ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:285:7: error: use of undeclared identifier 'CL_DEVICE_SHARED_SYSTEM_MEM_CAPABILITIES_INTEL'
      CL_DEVICE_SHARED_SYSTEM_MEM_CAPABILITIES_INTEL,
      ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:287:25: error: use of undeclared identifier 'CL_DEVICE_UUID_KHR'
  PI_DEVICE_INFO_UUID = CL_DEVICE_UUID_KHR,
                        ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:330:34: error: use of undeclared identifier 'CL_QUEUE_DEVICE_DEFAULT'
  PI_QUEUE_INFO_DEVICE_DEFAULT = CL_QUEUE_DEVICE_DEFAULT,
                                 ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:333:24: error: use of undeclared identifier 'CL_QUEUE_SIZE'
  PI_QUEUE_INFO_SIZE = CL_QUEUE_SIZE
                       ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:377:34: error: use of undeclared identifier 'CL_KERNEL_MAX_SUB_GROUP_SIZE_FOR_NDRANGE'
  PI_KERNEL_MAX_SUB_GROUP_SIZE = CL_KERNEL_MAX_SUB_GROUP_SIZE_FOR_NDRANGE,
                                 ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:378:34: error: use of undeclared identifier 'CL_KERNEL_MAX_NUM_SUB_GROUPS'; did you mean
      'PI_KERNEL_MAX_SUB_GROUP_SIZE'?
  PI_KERNEL_MAX_NUM_SUB_GROUPS = CL_KERNEL_MAX_NUM_SUB_GROUPS,
                                 ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:377:3: note: 'PI_KERNEL_MAX_SUB_GROUP_SIZE' declared here
  PI_KERNEL_MAX_SUB_GROUP_SIZE = CL_KERNEL_MAX_SUB_GROUP_SIZE_FOR_NDRANGE,
  ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:379:38: error: use of undeclared identifier 'CL_KERNEL_COMPILE_NUM_SUB_GROUPS'; did you mean
      'PI_KERNEL_MAX_NUM_SUB_GROUPS'?
  PI_KERNEL_COMPILE_NUM_SUB_GROUPS = CL_KERNEL_COMPILE_NUM_SUB_GROUPS,
                                     ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:378:3: note: 'PI_KERNEL_MAX_NUM_SUB_GROUPS' declared here
  PI_KERNEL_MAX_NUM_SUB_GROUPS = CL_KERNEL_MAX_NUM_SUB_GROUPS,
  ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:381:7: error: use of undeclared identifier 'CL_KERNEL_COMPILE_SUB_GROUP_SIZE_INTEL'
      CL_KERNEL_COMPILE_SUB_GROUP_SIZE_INTEL
      ^
/home/michel/sycl_workspace/deploy/include/sycl/CL/sycl/detail/pi.h:417:30: error: use of undeclared identifier 'CL_COMMAND_SVM_FREE'
  PI_COMMAND_TYPE_SVM_FREE = CL_COMMAND_SVM_FREE,
                             ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

Reply:
OK, it makes sense.